client/src/components/History/HistoryViews/JobHistoryModal.jsx

```diff
@@ -182,18 +214,33 @@ JobHistoryContent.propTypes = {
   // eslint-disable-next-line react/forbid-prop-types
   jobDisplay: PropTypes.object.isRequired,
   toggle: PropTypes.func.isRequired,
+  version: PropTypes.string.isRequired,
 };
 JobHistoryContent.defaultProps = {
   jobName: '',
   jobDisplay: {},
 };
 
-function JobHistoryModal({ uuid }) {
-  const loading = useSelector((state) => state.jobDetail.loading);
-  const loadingError = useSelector((state) => state.jobDetail.loadingError);
-  const { job, display } = useSelector((state) => state.jobDetail);
-  const { state } = useLocation();
-
+function JobHistoryModal({ uuid, version }) {
+  const { loading, loadingError, job, display } = useSelector((state) => {
+    if (version === 'v3') {
+      const jobv3 = state.jobDetail;
+      return {
+        loading: jobv3.loading,
+        loadingError: jobv3.loadingError,
+        job: jobv3.job,
+        display: jobv3.display,
+      };
+    } else {
+      const jobv2 = state.jobsv2.list.find((job) => job.id === uuid);
+      return {
+        loading: false,
+        loadingError: null,
+        job: jobv2,
+        display: jobv2.display,
+      };
+    }
+  });
```

Comment:
Maybe we can remove references to `v3`, that way when we remove the v2 jobs from the view later, we can just remove any v2 specific code more easily.